tracer/src/Datadog.Trace/ISpanContextGetter.cs

```diff
@@ -0,0 +1,29 @@
+// <copyright file="ISpanContextGetter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+#nullable enable
+
+namespace Datadog.Trace
+{
+    /// <summary>
+    /// The SpanContextGetter is responsible for extracting SpanContext in the rare cases where the Tracer couldn't propagate it itself.
+    /// This can happen for instance when libraries add an extra layer above the instrumented ones
+    /// (eg consuming Kafka messages and enqueuing them prior to generate a span).
+    /// </summary>
+    public interface ISpanContextGetter
```

Comment:
Not a fan of "getter" as a noun 😛 How about `ISpanContextExtractor`?

Reply:
Yes makes sense. It seems it's the naming used in[ OTel](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#context-interaction) as well.

While I'm at it, shall we name `ISpanContextGetter` to `ISpanContextPropagator` if we were to one day add injection (even though I don't see what use case that could cover)?

Reply:
>...if we were to one day add...

I'd say leave it as `ISpanContextExtractor`. We can add `ISpanContextInjector` if that day ever comes 😉

Reply:
Fixed in 1b1c81fbae0607ec30bccc9f65b44c1620b67a09